src/mcfret.c

```diff
@@ -0,0 +1,366 @@
+    vecr=(float *)calloc(non->singles*non->singles,sizeof(float));	
+    veci=(float *)calloc(non->singles*non->singles,sizeof(float));
+
+    /*Allocating memory for the density matrix*/
+    dens_mat=(float *)calloc(non->singles*non->singles,sizeof(float));
+
+ /* Open Trajectory files */
+    H_traj=fopen(non->energyFName,"rb");
+    if (H_traj==NULL){
+        printf("Hamiltonian file not found!\n");
+        exit(1);
+    }
+      /*Here we want to call the routine for checking the trajectory files*/ 
+    control(non);
+
+    itime=0;
+
+    N_samples=(non->length-non->tmax1-1)/non->sample+1;
+    if (N_samples>0) {
+        printf("Making %d samples!\n",N_samples);
+    } else {
+      printf("Insufficient data to calculate spectrum.\n");
+      printf("Please, lower max times or provide longer\n");
+      printf("trajectory.\n");
+      exit(1);
+    }
+
+    if (non->end==0) non->end=N_samples;
+    if (non->end>N_samples){
+      printf(RED "Endpoint larger than number of samples was specified.\n" RESET);
+      printf(RED "Endpoint was %d but cannot be larger than %d.\n" RESET,non->end,N_samples);
+      exit(0);
+    }
+
+    log=fopen("NISE.log","a");
+    fprintf(log,"Begin sample: %d, End sample: %d.\n",non->begin,non->end);
+    fclose(log);
+
+    
+    /* Read coupling: Since in this case couplings are static, we want to read them in only once*/
+    if (!strcmp(non->hamiltonian,"Coupling")){
+      C_traj=fopen(non->couplingFName,"rb");
+      if (C_traj==NULL){
+        printf("Coupling file not found!\n");
+        exit(1);
+      }
+      if (read_He(non,Hamil_i_e,C_traj,-1)!=1){
+        printf("Coupling trajectory file to short, could not fill buffer!!!\n");
+        exit(1);
+      }
+      fclose(C_traj);
+    }
+
+    /*Looping over samples: Each sample represents a different starting point on the Hamiltonian trajectory*/
+  
+    for (samples=non->begin;samples<non->end;samples++){ 
+        ti=samples*non->sample;
+        if (non->cluster!=-1){
+            if (read_cluster(non,ti,&cl,Cfile)!=1){
+	              printf("Cluster trajectory file to short, could not fill buffer!!!\n");
+	              printf("ITIME %d\n",ti);
+	              exit(1);
+            }
+            /*Configuration belong to cluster*/ 
+        if (non->cluster==cl){
+	          Ncl++;
+        }
+        }
+        if (non->cluster==-1 || non->cluster==cl){
+      /*Initialize time-evolution operator*/ 
+            unitmat(vecr,non->singles);
+      /*Do projection on selected sites if asked*/
+            if (non->Npsites>0){
+                for (i=0;i<non->singles;i++){
+                    j=i*non->singles;
+                    projection(vecr+j,non);
+                }
+            }
+            clearvec(veci,non->singles*non->singles);
+
+      /*Loop over delay*/ 
+            for (t1=0;t1<non->tmax;t1++){
+	              tj=ti+t1;
+	    /* Read Hamiltonian */
+	              if (!strcmp(non->hamiltonian,"Coupling")){
```

Comment:
The indentation is not consistent. Consider using a code linter for better readability. The indentation should be according to the style guide.